### PR TITLE
Add set testing

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds set testing.

Targets #181

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
This does not build. Can someone figure out why? I do not understand the error message here.

**Comment by neithernut** [comment]
The compiler doesn't get that the constant number you use for initializing the array is, indeed, constant.
Use a definition and you're fine.

**Comment by matthiasbeyer** [comment]
@neithernut Can you tell me why the build fails (it doesn't find the executeable somehow)?

**Comment by matthiasbeyer** [comment]
@neithernut I'm ready here. The PR includes fixes for the project sources as well, not only testing!

@TheNeikos review as well, please!

File: tests/check/objects/ws_set/test.c [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+static int
+compare_set_test_objs(
+    struct ws_object const* o1,
+    struct ws_object const* o2
+) {
+    if (o1 == o2) {
+        return 0;
+    } else {
+        return ((((void *) o1) > ((void *) o2)) ? -1 : 1);
```
Comment (TheNeikos) [inline tests/check/objects/ws_set/test.c]: you could use the signum macro instead of the if construct

Comment (matthiasbeyer) [inline tests/check/objects/ws_set/test.c]: Please provide an example for this, I can't wrap my slimy head around it atm.

Comment (neithernut) [inline tests/check/objects/ws_set/test.c]:
``` C
return signum(o2 - o1);
```

The macro is part of `util/arithmetical.h`

File: tests/check/objects/ws_set/test.c [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+static int
+compare_set_test_objs(
+    struct ws_object const* o1,
+    struct ws_object const* o2
+) {
+    if (o1 == o2) {
+        return 0;
+    } else {
+        return signum(o2 - o1);
+    }
```
Comment (neithernut) [inline tests/check/objects/ws_set/test.c]: You don't need the `if` at all. `signum(o2 - o1)` yields `0` if `o2 == o1`.

File: tests/check/objects/ws_set/test.c [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+    ck_assert(set != NULL);
+}
+
+static void
+test_set_setup_objs(void)
+{
+    test_set_setup();
+    int i;
+
+    for (i = N_TEST_OBJS - 1; i; --i) {
```
Comment (neithernut) [inline tests/check/objects/ws_set/test.c]: Either declare `i` inside the `for` or make it a `while`.

File: tests/check/objects/ws_set/test.c [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+    ck_assert(set_b == NULL);
+
+    set_a = ws_set_new();
+    set_b = ws_set_new();
+
+    ck_assert(set_a != NULL);
+    ck_assert(set_b != NULL);
+
+    int i;
+    for (i = N_TEST_OBJS - 1; i; --i) {
```
Comment (neithernut) [inline tests/check/objects/ws_set/test.c]: Same as above.

**Comment by matthiasbeyer** [comment]
@neithernut , @TheNeikos please review, this should go into the sources asap, as fixes are included here.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
No need to.

**Comment by neithernut** [comment]
Then ACK from me.

**Pull request merged**